lib/msf/core/post_mixin.rb

```diff
@@ -231,17 +231,20 @@ def session_incompatibility(sess_or_sid)
         return 'missing Meterpreter features: core can not be extended' unless s.commands.include?(Rex::Post::Meterpreter::COMMAND_ID_CORE_LOADLIB)
 
         # Since core is already loaded, if the missing command is a core command then it's truly missing
-        return 'missing Meterpreter features: core commands' if missing_cmd_ids.any? do |cmd_id|
+        missing_core_cmd_ids = missing_cmd_ids.select do |cmd_id|
           cmd_id.between?(
             Rex::Post::Meterpreter::ClientCore.extension_id,
             Rex::Post::Meterpreter::ClientCore.extension_id + Rex::Post::Meterpreter::COMMAND_ID_RANGE - 1
           )
         end
+        if missing_core_cmd_ids.any?
+          return "missing Meterpreter features: core commands - #{command_names_for(missing_core_cmd_ids)}"
```

Comment:
Since you're printing out the entire name of the missing command now, you don't really need to mention that they are core commands vs extension commands. That can easily be seen in the command name prefix. Removing that would keep the line a bit shorter and could look a little tidier.

```suggestion
          return "missing Meterpreter commands: #{command_names_for(missing_core_cmd_ids)}"
```

Reply:
Fixed :+1: